### DISCUSSION TITLE
Migrating from a Doctrine database to Mikro-ORM generates duplicated queries

### DISCUSSION THREAD

**Discussion by DavidGarciaCat** [mikro-orm/mikro-orm#3797] [General]
I have a functional Doctrine database, with several entities, mappings and indexes.

```php
#[ORM\Entity()]
#[ORM\Table(name: 'users')]
class User
{
    // ...
    #[ORM\Column(name: 'email', type: 'string', length: 100, unique: true, nullable: false)]
    private string $email;
    // ...
}
```

```
bin/console doctrine:migrations:diff
```

```php
$this->addSql('CREATE TABLE users ... UNIQUE INDEX UNIQ_1483A5E9E7927C74 (email) ...');
```

When I try to migrate the PHP app to a new TS app, the migration commands generate duplicated queries.

```
npx mikro-orm migration:create
```

```ts
    // ...
    this.addSql('alter table `users` drop index `UNIQ_1483A5E9E7927C74`;');
    // ...
    this.addSql('alter table `users` drop index `uniq_1483a5e9e7927c74`;');
    // ...
```

Shall we assume the issue is caused because the checks and query generation are based on an async process, thus some queries are being generated twice?

Is there any known way to fix this issue?

**Comment by B4nan** [comment]
Those queries are not the same, see the different casing: `uniq_` vs `UNIQ_`. It's weird that such queries were generated, I can look into that if you provide something reproducible (= your schema + entity definition + ORM config). This has nothing to do with anything being async.

Either way, doctrine has different default naming strategies for index names, once you run the first migration, there shouldn't be any more random queries.
